crates/ra_ide/src/inlay_hints.rs

```diff
@@ -75,8 +75,7 @@ fn get_chaining_hints(
         return None;
     }
 
-    let ty = sema.type_of_expr(&expr)?;
-    if ty.is_unknown() {
+    if expr.syntax().kind() == SyntaxKind::RECORD_LIT {
```

Comment:
```suggestion
    if matches!(expr, ast::Expr::RecordLit(_)) {
```
Not very like that caps enum ;0